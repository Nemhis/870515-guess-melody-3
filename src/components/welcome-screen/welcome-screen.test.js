import React from 'react';
import renderer from 'react-test-renderer';

import WelcomeScreen from './welcome-screen.jsx';

const name = `Boss`;

describe(`<WelcomeScreen>`, () => {
  it(`Render <WelcomeScreen>`, () => {
    const tree = renderer
      .create(<WelcomeScreen onWelcomeButtonClick={() => {}} />)
      .toJSON(`test`);

    expect(tree).toMatchSnapshot();
  });

  it(`Render <WelcomeScreen> with name`, () => {
    const tree = renderer
      .create(<WelcomeScreen
        name={name}
        onWelcomeButtonClick={() => {}}
      />)
      .toJSON(name);

    expect(tree).toMatchSnapshot();
  });
});
