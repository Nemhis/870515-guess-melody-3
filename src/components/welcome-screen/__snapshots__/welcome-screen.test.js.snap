// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeScreen> Render <WelcomeScreen> 1`] = `
Array [
  <p>
    Hello, 
    Stranger
    .
  </p>,
  <button
    className="welcome__button"
    onClick={[Function]}
  >
    <span
      className="visually-hidden"
    >
      Начать игру
    </span>
  </button>,
]
`;

exports[`<WelcomeScreen> Render <WelcomeScreen> with name 1`] = `
Array [
  <p>
    Hello, 
    Boss
    .
  </p>,
  <button
    className="welcome__button"
    onClick={[Function]}
  >
    <span
      className="visually-hidden"
    >
      Начать игру
    </span>
  </button>,
]
`;
